### DISCUSSION TITLE
Killed cluster via secrets-encryption?

### DISCUSSION THREAD

**Discussion by docbobo** [k3s-io/k3s#6043] [General]
So it seems as if I just killed my cluster by enabling secrets-encryption on an already running HA cluster. While the documentation seems contains lots of warnings regarding "following procedure", it just briefly mentions "NOTE: Starting K3s without encryption and enabling it at a later time is currently not supported." . Looks like "not supported" means corruption. Right now, all I am getting on startup of my server nodes is this:

> Aug 28 11:58:09 k3s-server-db0610 k3s[16097]: W0828 11:58:09.836071   16097 reflector.go:324] storage/cacher.go:/secrets: failed to list *core.Secret: illegal base64 data at input byte 3
> Aug 28 11:58:09 k3s-server-db0610 k3s[16097]: E0828 11:58:09.837246   16097 cacher.go:425] cacher (*core.Secret): unexpected ListAndWatch error: failed to list *core.Secret: illegal base64 data at input byte 3; reinitializing...
> Aug 28 11:58:10 k3s-server-db0610 k3s[16097]: W0828 11:58:10.865949   16097 reflector.go:324] storage/cacher.go:/secrets: failed to list *core.Secret: illegal base64 data at input byte 3
> Aug 28 11:58:10 k3s-server-db0610 k3s[16097]: E0828 11:58:10.866043   16097 cacher.go:425] cacher (*core.Secret): unexpected ListAndWatch error: failed to list *core.Secret: illegal base64 data at input byte 3; reinitializing...
> Aug 28 11:58:11 k3s-server-db0610 k3s[16097]: W0828 11:58:11.902610   16097 reflector.go:324] storage/cacher.go:/secrets: failed to list *core.Secret: illegal base64 data at input byte 3
> Aug 28 11:58:11 k3s-server-db0610 k3s[16097]: E0828 11:58:11.902719   16097 cacher.go:425] cacher (*core.Secret): unexpected ListAndWatch error: failed to list *core.Secret: illegal base64 data at input byte 3; reinitializing...

Before trying to reinitialize with some more or less recent snapshot, is there another way to fix this?

**Comment by brandond** [comment]
Yes, doing things that aren't supported may break your cluster. At this point I would confirm that it's disabled, and restore from a backup.

**Comment by docbobo** [comment]
Okay, that's pretty much what I ended up doing. I still believe that maybe this should be a little bit more "in the face" as this rather minor "info" field can be easily overseen.

That aside - is there any way to enable secrety-encryption in my scenario? E.g. when recreating the cluster from the snapshot?

**Reply by brandond** [reply]
You would need to manage your own secrets encryption configuration file, as described here: https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/
